internal/access/organization.go

```diff
@@ -0,0 +1,62 @@
+package access
+
+import (
+	"github.com/gin-gonic/gin"
+
+	"github.com/infrahq/infra/internal/server/data"
+	"github.com/infrahq/infra/internal/server/models"
+	"github.com/infrahq/infra/uid"
+)
+
+func ListOrganizations(c *gin.Context, name string, pg *models.Pagination) ([]models.Organization, error) {
+	var selectors = []data.SelectorFunc{}
+	if name != "" {
+		selectors = append(selectors, data.ByName(name))
+	}
+
+	roles := []string{models.InfraAdminRole}
+	db, err := RequireInfraRole(c, roles...)
+	if err == nil {
+		return data.ListOrganizations(db, pg, selectors...)
+	}
+	err = HandleAuthErr(err, "organizations", "list", roles...)
+
+	// TODO:
+	//    * Consider allowing a user to list their own organization
+
+	return nil, err
+}
+
+func GetOrganization(c *gin.Context, id uid.ID) (*models.Organization, error) {
+	roles := []string{models.InfraAdminRole}
+	db, err := RequireInfraRole(c, roles...)
+	if err != nil {
+		return nil, HandleAuthErr(err, "organizations", "get", roles...)
+	}
+
+	return data.GetOrganization(db, data.ByID(id))
+}
+
+func CreateOrganization(c *gin.Context, org *models.Organization) error {
+	roles := []string{models.InfraAdminRole}
+	db, err := RequireInfraRole(c, roles...)
```

Comment:
probably anyone should be able to create an organization, but maybe that's through a specific signup flow and not through this function?

Reply:
Yes. There's another commit which adds the `InfraSupportRole` which this will be modified to use in the future, and most people will instead use `/api/signup` to create the org.